variations.py

```diff
@@ -1,14 +1,49 @@
 from copy import deepcopy
 
 from .booking import Unit
+from .booking import dataset_from_crownoutput
 from .booking import dataset_from_artusoutput
 from .utils import Selection
 from .utils import Variation
 
 import logging
 logger = logging.getLogger(__name__)
 
+class ChangeDatasetCrown(Variation):
+    """
+    Variation that with the method create makes a deepcopy of
+    the selections and actions inside the unit passed as argument and substitutes
+    the directory attribute with the according variation.
+
+    Args:
+        name (str): name used to identify the instance of
+            this class
+        variation (str)
+    """
+    def __init__(self,
+            name, variation):
+        Variation.__init__(self, name)
+        self.variation = variation
 
+    def create(self, unit):
+        new_selections = deepcopy(unit.selections)
+        new_actions = deepcopy(unit.actions)
+        if self.variation not in unit.dataset.quantities_per_vars:
+            logger.fatal('Variation {} not found in ntuple'.format(self.variation))
+            raise NameError
+        else:
+            list_of_quants = unit.dataset.quantities_per_vars[self.variation]
+            for quant_i in list_of_quants:
+                for cut_i in range(len(new_selections[0].cuts)):
+                    if quant_i in new_selections[0].cuts[cut_i].expression:
+                        new_selections[0].cuts[cut_i].expression = new_selections[0].cuts[cut_i].expression.replace(quant_i, "{quant}__{var}".format(quant=quant_i, var=self.variation))
+                for weight_i in range(len(new_selections[1].weights)):
+                    if quant_i in new_selections[1].weights[weight_i].expression:
```

Comment:
I think the selection switch based on the number of objects in the selection list will not work properly. It might work for the case you are checking at the moment but for a different process or the analysis categorisation it will probably not work.

Reply:
In general, you have multiple selection objects and in principle each of them could contain of cuts and weights.